### DISCUSSION TITLE
flowtype/no-internal-flow-type eslint error for flow-typed

### DISCUSSION THREAD

**Discussion by jcolakk** [adeira/universe#2079] [Q&A]
Hi 👋  in our repo we started to get `flowtype/no-internal-flow-type` eslint error for package types in `flow-typed`.  We are using `@adeira/eslint-config` in our repo.
How do you recommend handling this?

1. Ignore `flow-typed` in .eslintignore
2. Change it to not use internal flow type, e.g from `React$Element` to `ReactElement`, but this causes flow error instead `Cannot resolve name ReactElement`

**Comment by tbergquist-godaddy** [comment]
```
import * as React from 'react';

let Component: React.Element<{children: React.Node}>;
```

**Reply by mrtnzlml** [reply]
@tbergq Not sure it's a good idea to modify 3rd party `flow-typed` definitions like this. Maybe rather update the upstream repo? 🤔

**Comment by mrtnzlml** [comment]
It's strange that you started getting some new errors since the only thing which changed was the rule name, see: https://github.com/adeira/universe/commit/e624e72cdd4fa373fe3c01830cd3fc8237e07187 (Maybe you had `adeira/no-internal-flow-type` rule ignored? Search for it in your codebase.)

Anyway, in case of this repository, we are [not running lints for `flow-typed` directory](https://github.com/adeira/universe/blob/da1ea59be9c285635341706b33685ae8e7b321cf/.jest-eslint.config.js#L10) because the assumption is that the code is OK and not edited manually very often since it's mostly 3rd party.

I'd recommended you to ignore it as well (or at least `flow-typed/npm`).

**Reply by tbergquist-godaddy** [reply]
Sorry, I missed that this was in flow-typed 🙈

**Reply by jcolakk** [reply]
Ah, yes, we had it disabled for flow-typed already. That makes sense, I also didn't want to update the 3rd party types.
Thanks guys 👍

**Reply by mrtnzlml** [reply]
Sorry for the breaking change. I didn't realize it could have such consequences.
